modules/transactions/src/main/java/org/apache/ignite/internal/tx/InternalTransaction.java

```diff
@@ -117,9 +117,10 @@ void enlist(
      * @param executionTimestamp The timestamp is the time when a read-only transaction is applied to the remote node. The parameter
      *         is not used for read-write transactions.
      * @param full Full state transaction marker.
+     * @param timeoutExceeded Timeout exceeded flag (commit flag must be {@code false}).
      * @return The future.
      */
-    CompletableFuture<Void> finish(boolean commit, @Nullable HybridTimestamp executionTimestamp, boolean full);
+    CompletableFuture<Void> finish(boolean commit, @Nullable HybridTimestamp executionTimestamp, boolean full, boolean timeoutExceeded);
```

Comment:
A transaction might be aborted for different reasons, at least

1. user request
2. an operation error
3. a timeout
4. schema change

We currently don't record 1 and 2, and 4 is not implemented (but we might implement it later). 3 is the only abort reason that stands out now, but should we still represent it using an enum and not a boolean flag?

Reply:
This is a very good proposal. Do you think I should implement this in current PR or create a ticket?

Reply:
https://issues.apache.org/jira/browse/IGNITE-24727